daemon/graphdriver/quota/projectquota.go

```diff
@@ -194,6 +247,49 @@ func setProjectQuota(backingFsBlockDev string, projectID uint32, quota Quota) er
 	return nil
 }
 
+// Ext4Quota is a struct implements quota operations
+type Ext4Quota struct {
+}
+
+// SetProjectQuota - set the quota for project id on ext4 block device
+func (q *Ext4Quota) SetProjectQuota(backingFsBlockDev string, projectID uint32, quota Quota) error {
```

Comment:
Am I wrong to assume that this is not really the "Ext4Quota" implementation, but the new generic "VfsQuota" implementation, which can also be applied to other fs that may support project quota in the future?
Am I wrong to assume that since kernel version v4.5, both xfs and ext4 supports that same unified API?
Am I wrong to assume that if you need to use  Q_SETPQUOTA, you also need to use Q_GETPQUOTA and Q_GETPQSTAT for ext4?
If I am not wrong, I suggest to determine the QuotafileOps to use by kernel version (either XFS legacy or VFS ops) and to include also GetProjectQuota() and GetProjectQuotaStat() in the QuotafileOps struct.